src/algebra/algebra/basic.lean

```diff
@@ -1248,13 +1251,12 @@ def to_int_alg_hom [ring R] [ring S] [algebra ℤ R] [algebra ℤ S] (f : R →+
   R →ₐ[ℤ] S :=
 { commutes' := λ n, by simp, .. f }
 
-@[simp] lemma map_rat_algebra_map [ring R] [ring S] [algebra ℚ R] [algebra ℚ S] (f : R →+* S)
-  (r : ℚ) :
-  f (algebra_map ℚ R r) = algebra_map ℚ S r :=
+@[simp] lemma map_rat_algebra_map [semiring R] [semiring S] [algebra ℚ R] [algebra ℚ S]
```

Comment:
This change doesn't look useful to me, every rat-algebra must already be a ring

Reply:
Sure, but there's no reason not to do it because at worst it prevents a letI upgrading things to rings l

Reply:
I can't imagine that `letI` ever happening, except in a proof already full of `letI`s.

One reason not to do it is that typeclass search has to work harder for this spelling; even though in practice a `ring` will always be in scope, this forces lean to first look for semiring instances.

The main argument I'd use though is that the generalization has no mathematical content, and just makes things more confusing for a human reading it.

Reply:
I'll roll this back, then!